### PR TITLE
time-varying params

### PR DESCRIPTION
**Pull request by blanchco**
# Description

Initial implementation of time varying params in model config:

https://github.com/DARPA-ASKEM/terarium/assets/33158416/0b4acd27-19b8-4889-b836-05f55c50fd6f

### CODE REVIEW

File: packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue [hunk 1]
```diff
@@ -165,26 +181,80 @@ const openMatrixModal = (datum: ModelConfigTableData) => {
...
+			isValid = false;
+		}
+	});
+
+	errorMessage.value = '';
+	if (!isValid) {
+		errorMessage.value = 'Incorrect Format (e.g., 0:500, 10:550, 25:700 etc)';
+	}
+	return isValid;
+};
```
Comment (YohannParis) [inline packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue:221]: You could simplify a lot this function now that you defined it.

```suggestion
const validateTimeSeries = (values: string): boolean => {
    const message = 'Incorrect Format (e.g., 0:500, 10:550, 25:700 etc)';
    
    if (typeof values !== 'string') {
        errorMessage.value = message;
        return false;
    }

    const isPairValid = (pair: string): boolean => /^\d+:\d+$/.test(pair.trim());
    const isValid = values.split(',').every(isPairValid);

    errorMessage.value = isValid ? '' : message;
    return isValid;
};
```

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue [hunk 2]
```diff
@@ -165,26 +181,80 @@ const openMatrixModal = (datum: ModelConfigTableData) => {
...
+			isValid = false;
+			return;
+		}
+
+		const [time, value] = pair.split(':').map((part) => part.trim());
+
+		if (!/^\d+$/.test(time) || !/^\d+(\.\d+)?$/.test(value)) {
+			isValid = false;
+		}
+	});
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue:214]: FYI: This forEach is a bit inefficient, you will keep on parsing even if isValid is false

File: packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue (hunk 1 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue]: changed

File: packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue (hunk 2 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue]: changed in favour for the function yohann provided

**Pull request merged**